### ISSUE TITLE
Missing template type warning under NTI

### ISSUE THREAD

**Issue by chrisdoble** [google/closure-compiler#2020]
Please consider the following code (a contrived simplification of the code in which I found this issue):

``` js
// enum.js
/** @enum {string} */
export const Enum = {VALUE: 'VALUE'};
```

``` js
// test.js
import { Enum } from './enum';

/**
 * @template T
 * @param {T|null|undefined} val
 * @param {T} def
 * @return {T}
 */
function getOrDefault(val, def) {
  return val != null ? val : def;
}

/** @param {Enum} e */
function logEnum(e) {
  console.log(e);
}

(function() {
  logEnum(getOrDefault('', Enum.VALUE));
}());
```

Compiling this under OTI produces a warning:

```
$ java -jar compiler-latest/closure-compiler-v20160911.jar --checks_only --warning_level VERBOSE enum.js test.js
test.js:19: WARNING - actual parameter 1 of logEnum$$module$test does not match formal parameter
found   : string
required: Enum$$module$enum<string>
  logEnum(getOrDefault('', Enum.VALUE));
          ^

0 error(s), 1 warning(s), 85.71428571428571% typed
$
```

Compiling it under NTI does not:

```
$ java -jar compiler-latest/closure-compiler-v20160911.jar --checks_only --new_type_inf --warning_level VERBOSE enum.js test.js
$
```

A warning is produced under NTI if:
- `Enum` is inlined in `test.js` (i.e. the `import` is removed), or
- the IIFE around the `logEnum` call is removed.

Is this behaviour correct? I expected a warning to be produced under NTI.

**Comment by dimvar** [comment]
This seems to be related to how we transpile ES6 modules. With a `goog.module`, NTI finds the warning.

https://closure-compiler-debugger.appspot.com/#input0%3Dgoog.module('Enum')%253B%250A%250A%252F**%2520%2540enum%2520%257Bstring%257D%2520*%252F%250Avar%2520Enum%2520%253D%2520%257BVALUE%253A%2520'VALUE'%257D%253B%250A%250Aexports%2520%253D%2520Enum%253B%26input1%3Dgoog.module('foobar')%253B%250A%250Avar%2520Enum%2520%253D%2520goog.require('Enum')%253B%250A%250A%252F**%250A%2520*%2520%2540template%2520T%250A%2520*%2520%2540param%2520%257BT%257Cnull%257Cundefined%257D%2520val%250A%2520*%2520%2540param%2520%257BT%257D%2520def%250A%2520*%2520%2540return%2520%257BT%257D%250A%2520*%252F%250Afunction%2520getOrDefault(val%252C%2520def)%2520%257B%250A%2520%2520return%2520val%2520!%253D%2520null%2520%253F%2520val%2520%253A%2520def%253B%250A%257D%250A%250A%252F**%2520%2540param%2520%257BEnum%257D%2520e%2520*%252F%250Afunction%2520logEnum(e)%2520%257B%250A%2520%2520console.log(e)%253B%250A%257D%250A%250A(function()%2520%257B%250A%2520%2520logEnum(getOrDefault(''%252C%2520Enum.VALUE))%253B%250A%257D())%253B%250A%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1
